cmd/update-useragent/main.go

```
@@ -1,190 +1,189 @@
+	browsers      = []string{"chrome", "edge", "safari", "firefox", "firefox-esr"}
 )
 
 func main() {
-	// load user agents from
 	flag.Parse()
 
-	uas, err := parseSources()
+	userAgents := getUserAgents()
+	data, err := json.Marshal(userAgents)
 	if err != nil {
 		log.Fatal(err)
 	}
 
-	data, err := json.Marshal(uas)
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	// dump to output file
 	err = os.WriteFile(*userAgentData, data, 0644)
 	if err != nil {
 		log.Fatal(err)
 	}
 
 }
 
-// getHttpClient with common options
-func getHttpClient() *http.Client {
-	transport := http.Transport{
-		Proxy: http.ProxyFromEnvironment,
-		TLSClientConfig: &tls.Config{
-			InsecureSkipVerify: true,
-		},
-	}
-
-	return &http.Client{Transport: &transport}
-}
-
-// parseSources and retrieve a list of user agents with metadata
-func parseSources() ([]*useragent.UserAgent, error) {
-	uas1, err := parseTechPattern()
-	if err != nil {
-		return nil, err
-	}
-	uas2, err := parseGhRepo()
-	if err != nil {
-		return nil, err
-	}
-
-	return append(uas1, uas2...), nil
-}
-
-const UrlTechPattern = "https://techpatterns.com/downloads/firefox/useragentswitcher.xml"
-
-// Parses source at "https://techpatterns.com/downloads/firefox/useragentswitcher.xml"
-func parseTechPattern() ([]*useragent.UserAgent, error) {
-	httpClient := getHttpClient()
+func getUserAgents() []*useragent.UserAgent {
+	var userAgents []*useragent.UserAgent
 
-	resp, err := httpClient.Get(UrlTechPattern)
-	if err != nil {
-		return nil, err
-	}
+	httpClient := buildHttpClient()
+	for _, browser := range browsers {
+		whatismybrowserApiUrl := fmt.Sprintf(`https://api.whatismybrowser.com/api/v2/user_agent_database_search?order_by=times_seen%%20desc&limit=300&offset=500&software_name=%s`, browser)
+		req, err := http.NewRequest("GET", whatismybrowserApiUrl, nil)
+		if err != nil {
+			continue
+		}
+		req.Header.Add("X-API-KEY", os.Getenv("API_KEY"))
```

we can replace `API_KEY` with `WHATISMYBROWSER_KEY` and fail to run if env is not available, for now it fails silently and write `null` to UA data.